compiler/rustc_resolve/src/lib.rs

```diff
@@ -2584,10 +2584,25 @@ impl<'a> Resolver<'a> {
                                 res_err = Some(CannotCaptureDynamicEnvironmentInFnItem);
                             }
                         }
-                        ConstantItemRibKind(_) => {
+                        ConstantItemRibKind(_, ident) => {
                             // Still doesn't deal with upvars
                             if record_used {
-                                self.report_error(span, AttemptToUseNonConstantValueInConstant);
+                                if let Some(ident) = ident {
+                                    let sugg = "consider using `let` instead of `const`";
+                                    self.report_error(
+                                        span,
+                                        AttemptToUseNonConstantValueInConstant(Some(ident), sugg),
+                                    );
+                                } else {
+                                    let sugg = "consider using `const` instead of `let`";
+                                    self.report_error(
+                                        rib_ident.span,
+                                        AttemptToUseNonConstantValueInConstant(
+                                            Some(rib_ident),
+                                            sugg,
+                                        ),
+                                    );
```

Comment:
The current output is not good but I couldn't find the way to find the span of `n` definition from Resolver. How to implement that?

```
     |         let n = 42;
4 LL |         let _x = [0; n];
+    |                      |
+    |                      non-constant value
+    |                      help: consider using `const` instead of `let`
```

I want to output like following.

```
     |         let n = 42;
+    |             - help: consider using `const` instead of `let`
4 LL |         let _x = [0; n];
+    |                      |
+    |                      non-constant value
```